lib/widgets/message_list.dart

```diff
@@ -635,3 +702,75 @@ final _kMessageTimestampStyle = TextStyle(
   fontSize: 12,
   fontWeight: FontWeight.w400,
   color: const HSLColor.fromAHSL(0.4, 0, 0, 0.2).toColor());
+
+Future<void> markNarrowAsRead(BuildContext context, ApiConnection connection, Narrow narrow) async {
+  final zulipLocalizations = ZulipLocalizations.of(context);
+  // Use [AnchorCode.oldest], because [AnchorCode.firstUnread]
+  // will be the oldest non-muted unread message, which would
+  // result in muted unreads older than the first unread not
+  // being processed.
+  Anchor anchor = AnchorCode.oldest;
+  int responseCount = 0;
+  int updatedCount = 0;
+  while (true) {
+    final result = await updateMessageFlagsForNarrow(connection,
+      anchor: anchor,
+      // anchor="oldest" is an anchor ID lower than any valid
```

Comment:
"[AnchorCode.oldest]" — 'anchor="oldest"' doesn't mean much in a Dart context

(For that matter it doesn't mean much in a JS context — it looks like whoever wrote the JS comment you're copying here was thinking in Python.)